File: src/python_example/classification/beit_large_patch16/beit_large_patch16_async.py
#!/usr/bin/env python3
# Copyright (C) 2018- DEEPX Ltd. All rights reserved.
"""
Beit large patch16 Asynchronous Inference Example

Usage:
    python beit_large_patch16_async.py --model model.dxnn --image input.jpg
"""

import sys
from pathlib import Path

_module_dir = Path(__file__).parent
_v3_dir = _module_dir.parent.parent
for _path in [str(_v3_dir), str(_module_dir)]:
    if _path not in sys.path:
        sys.path.insert(0, _path)

from factory import Beit_large_patch16Factory
from common.runner import AsyncRunner, parse_common_args


def parse_args():
    return parse_common_args("Beit large patch16 Async Inference")


def main():
    args = parse_args()
    factory = Beit_large_patch16Factory()
    runner = AsyncRunner(factory)
    runner.run(args)


if __name__ == "__main__":
    main()

File: src/cpp_example/classification/beit_large_patch16/beit_large_patch16_async.cpp
/**
 * @file beit_large_p16_async.cpp
 * @brief BeitLargeP16Factory asynchronous inference example
 */

#include "factory/beit_large_patch16_factory.hpp"
#include "common/runner/async_classification_runner.hpp"

int main(int argc, char* argv[]) {
    auto factory = std::make_unique<dxapp::BeitLargeP16Factory>();
    dxapp::AsyncClassificationRunner<dxapp::BeitLargeP16Factory> runner(std::move(factory));
    return runner.run(argc, argv);
}
